### ISSUE TITLE
IndexError: pop from empty list

### ISSUE THREAD

**Issue by blashmet** [duo-labs/cloudmapper#256]
Verified AWS IAM permissions, but running "python cloudmapper.py collect --account accountname" returns the following error (after successfully grabbing a significant portion of the account metadata):  

File "/Users/user/Documents/Tools/cloudmapper/commands/collect.py", line 66, in call_function
    check = check.pop()
IndexError: pop from empty list

**Comment by 0xdabbad00** [comment]
Hmm, this is related to new functionality added yesterday.  That line is here: https://github.com/duo-labs/cloudmapper/blob/34c54daed8c1434a4ee51a1930f49db014f0f206/commands/collect.py#L66

It only hits for the Access Advisor data collection here: https://github.com/duo-labs/cloudmapper/blob/34c54daed8c1434a4ee51a1930f49db014f0f206/collect_commands.yaml#L43

What should be happening at that `check.pop()` line is  is `check` should be equal to `[{"Name": "JobStatus", "Value": "COMPLETED"}]` so I'm just trying to grab that first value out of the array. 

I'm going to assume you didn't modify the files at all.  Try putting a `print` above that `check = check.pop()` line to see what it says.  It would also be helpful to know what data was being collected then. It should be `iam:get-service-last-accessed-details`

**Comment by blashmet** [comment]
Thanks for the response.

The data that you expected to be under collection is correct (i.e., iam:get-service-last-accessed-details).

It looks like `check` has the correct values during an initial evaluation, but is then empty during a second one (see code below).

Weird because I would expect `if check is not None:` to prevent this.

I tried changing to `if check` and it seems to get past the point where it was failing.

Any issues with doing that?


`Response already collected at account-data/./account/us-east-1/iam-get-service-last-accessed-details/88b3f34a-7341-5b1e-a9c7-fdfaba49186b`
`printing check array for debug...`
`[{'Name': 'JobStatus', 'Value': 'COMPLETED'}]`
 `Making call for account-data/./account/us-east-1/iam-get-service-last-accessed-details/3cb3f34a-58a4-9a26-72a4-8bdceb819c61`
`printing check array for debug...`
[]
`Traceback (most recent call last):`
 ` File "cloudmapper.py", line 69, in <module>
    main()`
  `File "cloudmapper.py", line 64, in main
    commands[command].run(arguments)`
  `File "/Users/user/Documents/Tools/cloudmapper/commands/collect.py", line 306, in run
    collect(args)`
  `File "/Users/user/Documents/Tools/cloudmapper/commands/collect.py", line 260, in collect
    summary)`
  `File "/Users/user/Documents/Tools/cloudmapper/commands/collect.py", line 68, in call_function
    check = check.pop()`
`IndexError: pop from empty list`

**Comment by 0xdabbad00** [comment]
Seems like something unintended is happening related to things being passed by reference as opposed to pass by value that it seems I expected.

I probably need to remove the lines
```
    if check is not None:
        check = check.pop()
```
at https://github.com/duo-labs/cloudmapper/blob/34c54daed8c1434a4ee51a1930f49db014f0f206/commands/collect.py#L65

and then change:
```
if check is not None:
                if data[check['Name']] == check['Value']:
```
to
```
if check is not None:
                if data[check[0]['Name']] == check[0]['Value']:
```

**Comment by 0xdabbad00** [comment]
This should be resolved now in #258
